### PR TITLE
Check for duplicate Reference IDs on Frame and Scene creation

### PR DESCRIPTION
**Pull request by jean-lucas**
As title suggests.

The original call for duplicates was misplaced, and not being called.

### CODE REVIEW

File: nucleus/dataset.py [hunk 1]
```diff
@@ -572,8 +575,6 @@ def append(
                 video_scenes, update, asynchronous
             )
 
-        check_for_duplicate_reference_ids(dataset_items)
-
```
Comment (jean-lucas) [inline nucleus/dataset.py:576]: this was not being checked for Scenes + Video

File: nucleus/scene.py [hunk 2]
```diff
@@ -38,16 +42,15 @@ class Frame:
     """
 
     def __init__(self, **kwargs):
-        self.items = {}
+        self.items: Dict[str, DatasetItem] = {}
         for key, value in kwargs.items():
-            self.items[key] = value
-
-    def __post_init__(self):
```
Comment (jean-lucas) [inline nucleus/scene.py:45]: __post_init__ does not work on default classes.

File: nucleus/dataset_item.py [hunk 3]
```diff
@@ -195,13 +195,15 @@ def check_all_paths_remote(dataset_items: Sequence[DatasetItem]):
 def check_for_duplicate_reference_ids(dataset_items: Sequence[DatasetItem]):
     ref_ids = []
     for dataset_item in dataset_items:
-        if dataset_item.reference_id is not None:
-            ref_ids.append(dataset_item.reference_id)
+        if dataset_item.reference_id is None:
+            raise ValueError("Reference ID cannot be None")
```
Comment (drakejwong) [inline nucleus/dataset_item.py:199]:
```suggestion
            raise ValueError(f"Reference ID cannot be None! Encountered DatasetItem with no reference ID:\n{dataset_item}")
```
maybe something like this to make it easier for the user to debug? assuming the stack trace won't show it

File: nucleus/scene.py (hunk 2 above)
Comment (drakejwong) [reply nucleus/scene.py:45]: nice catch!!

**Review by drakejwong — APPROVED** [review]
ty! just one nit to improve error logging

**Pull request merged**